pkg/skaffold/util/store.go

```diff
@@ -54,6 +54,15 @@ func (o *SyncStore) Exec(key string, f func() interface{}) interface{} {
 	return val
 }
 
+// Store will store the results for a key in a cache
+// It makes sure only one execution is in-flight for a given key at any time.
+func (o *SyncStore) Store(key string, r interface{}) {
+	o.sf.Do(fmt.Sprintf("%s-put", key), func() (interface{}, error) {
```

Comment:
I think it's some sort of anti-pattern to call `sf.Do` with different function callbacks. Also maybe it can cause a bug since it's not guaranteed which function result will be returned if there are multiple goroutines calling different functions. WDYT?

Reply:
The `singleflight.Do` documentation does not say anything like that. Looking at the code, it based on the "key" string.

The reason, i have it here is because, `sync.Map` does not support reading/writing multiple keys across subroutines.

The chances of `Store` for an artifact key being called across multiple go subroutines are 0. i can drop the `sf.Do` here.

Reply:
I totally missed the different key. You're right. We can do without it.